lib/java_buildpack/container/tomcat/tomcat_geode_store.rb

```diff
@@ -0,0 +1,168 @@
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/container'
+require 'java_buildpack/container/tomcat/tomcat_utils'
+require 'java_buildpack/logging/logger_factory'
+
+module JavaBuildpack
+  module Container
+
+    # Encapsulates the detect, compile, and release functionality for Tomcat Redis support.
+    class TomcatGeodeStore < JavaBuildpack::Component::VersionedDependencyComponent
+      include JavaBuildpack::Container
+
+      # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
+      def supports?
+        @application.services.one_service? FILTER, KEY_LOCATORS, KEY_USERS
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        return unless supports?
+        credentials = @application.services.find_service(FILTER)['credentials']
+        user = credentials[KEY_USERS].find { |u| u['username'] == 'cluster_operator' }
+
+        @droplet.java_opts.add_system_property 'gemfire.security-username', 'cluster_operator'
+        @droplet.java_opts.add_system_property 'gemfire.security-password', user['password']
+        @droplet.java_opts.add_system_property 'gemfire.security-client-auth-init',
+                                               'io.pivotal.cloudcache.ClientAuthInitialize.create'
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        return unless supports?
```

Comment:
No need for this (that's why you don't see it anywhere else).  `compile` isn't called unless `detect` first confirms its use.